lstchain/scripts/lstchain_dl1_to_dl2.py

```diff
@@ -141,13 +140,21 @@ def main():
     # Source-dependent analysis
     if config['source_dependent']:
 
+        # if source-dependent parameters are already in dl1 data, just read those data.
+        if dl1_params_src_dep_lstcam_key in get_dataset_keys(args.input_file):
+            data_srcdep = get_srcdep_params(args.input_file)
+
+        # if not, source-dependent parameters are added now
+        else:
+            data_srcdep = pd.concat(dl1_to_dl2.get_source_dependent_parameters(
+                data, config, focal_length=focal_length), axis=1)
+
         dl2_srcdep_dict = {}
         srcindep_keys = data.keys()
-        srcdep_index_keys = get_srcdep_index_keys(args.input_file)
+        srcdep_index_keys = data_srcdep.columns.levels[0]
```

Comment:
why are these read in a different way as the source independent ones?

Reply:
it was a bit confusing indeed...
On this line, assumed wobble angles, not parameter keys,  are loaded from source-dep data frame.
Those angles are saved as level 0 of the data frame, and each parameter key is saved as level 1 of the data.
I modified the name of this parameter (srcdep_index_keys -> srcdep_wobble_angles).

Reply:
but is this really wobble angles or off positions? we could take data with 2 or 4 different wobble angles and take calculate the off src-dependent parameter from 1, 3, 5 or any number of OFF positions

Reply:
ok, you're right. It should be independent of wobble angle (observation conditions).
It would be better to use 'srcdep_assumed_positions' or something?
This column contains 'on', 'off_090', 'off_180', 'off_270' for example in case that n_off=3 is assumed.

Reply:
yes, `srcdep_assumed_positions` sounds good

Reply:
ok, changed.